Tribler/Policies/BoostingSource.py

```diff
@@ -0,0 +1,639 @@
+                            channel_dict[channel.id] = channel
+
+                    # creating torrents
+                    torrents = []
+                    for hit in tor_values:
+                        if hit:
+                            chan_torrent = ChannelTorrent(*hit[1:] + [channel_dict.get(hit[0], None), None])
+                            chan_torrent.torrent_db = self.boosting_manager.torrent_db
+                            chan_torrent.channelcast_db = self.channelcast_db
+
+                            if chan_torrent.name:
+                                torrents.append(chan_torrent)
+
+                    return torrents
+
+                listtor = create_torrents(torrent_values, True,
+                                          {self.channel_id: self.channelcast_db.getChannel(self.channel_id)})
+                # listtor = self.gui_util.channelsearch_manager._createTorrents(
+                #     torrent_values, True, {self.channel_id: self.channelcast_db.getChannel(self.channel_id)})[2]
+
+                # dict {key_infohash(binary):Torrent(object-GUIDBTuple)}
+                self.unavail_torrent.update({t.infohash: t for t in listtor if t.infohash not in self.torrents})
+
+                # it's highly probable the checktor function is running at this time (if it's already running)
+                # if not running, start the checker
+                if not self.boosting_manager.is_pending_task_active(hexlify(self.source) + "_checktor"):
+                    self.boosting_manager.register_task(hexlify(self.source) + "_checktor",
+                                                        LoopingCall(self._check_tor)).start(10, now=True)
+
+    def _on_database_updated(self, subject, change_type, infohash):
+        if (subject, change_type, infohash) is None:
+            # Unused arguments
+            pass
+        self.database_updated = True
+
+    def get_source(self):
+        return self.channel.name if self.channel else None
+
+
+class RSSFeedSource(BoostingSource):
+    """
+    Credit mining source from a RSS feed.
+
+    # supported list (tested) :
+    # http://bt.etree.org/rss/bt_etree_org.rdf
+    # http://www.mininova.org/rss.xml
+    # https://kat.cr (via torcache)
+
+    # not supported (till now)
+    # https://eztv.ag/ezrss.xml (https link)
+    """
+
+    def __init__(self, session, rss_feed, boost_settings, callback):
+        BoostingSource.__init__(self, session, rss_feed, boost_settings, callback)
+
+        self.unescape = HTMLParser.HTMLParser().unescape
+
+        self.feed_handle = None
+
+        self.session.lm.threadpool.add_task(lambda feed=rss_feed: self._load_if_ready(feed), 0,
+                                            task_name=str(self.source) + "_load")
+
+        self.title = ""
+        self.description = ""
+        self.total_torrents = 0
+
+    def _load(self, rss_feed):
+        self.feed_handle = self.session.lm.ltmgr.get_session().add_feed(
```

Comment:
I didn't know libtorrent supports adding RSS feeds! @whirm maybe we should also use this for our channel manager?